### PR TITLE
Implement `bytes.__bytes__`

### PR DESCRIPTION
**Pull request by Snowapril**
This revision implement `bytes.__bytes__` in cpython 3.11 for typing module.

### References
* https://docs.python.org/3.11/whatsnew/3.11.html#other-cpython-implementation-changes
* https://bugs.python.org/issue24234

### CODE REVIEW

File: vm/src/builtins/bytes.rs [hunk 1]
```diff
@@ -128,6 +128,11 @@ impl PyBytes {
         &self.inner.elements
     }
 
+    #[pymethod(name = "__bytes__")]
+    pub fn bytes(&self) -> Vec<u8> {
+        self.inner.elements.to_vec()
+    }
+
```
Comment (qingshi163) [inline vm/src/builtins/bytes.rs:135]: If the object itself is an exact PyBytes, we should return the reference of the object instead creating a new object.

**Review by qingshi163 — CHANGES_REQUESTED** [review]

File: vm/src/builtins/bytes.rs (hunk 1 above)
Comment (Snowapril) [reply vm/src/builtins/bytes.rs:135]: Is this right? 
```rust
#[pymethod(magic)]
pub fn bytes(zelf: PyRef<Self>) -> PyRef<Self> {
    zelf
}
```

Comment (Snowapril) [reply vm/src/builtins/bytes.rs:135]: I suffered from exactly same problem in #2984. When I return PyBytes itself, below test go failed.
```python
class bytes_subclass(bytes):
    pass

bar = bytes_subclass(b'bar')
self.assertEqual(bar.__bytes__(), bar)
self.assertEqual(type(bar.__bytes__()), self.type2test)
```

Comment (Snowapril) [reply vm/src/builtins/bytes.rs:135]: Ah I'm sorry. I understood. Let me try one more time

Comment (Snowapril) [reply vm/src/builtins/bytes.rs]: @qingshi163 I applied it! Could you check one more time please? Thanks 😊

File: vm/src/builtins/bytes.rs [hunk 2]
```diff
@@ -128,6 +128,15 @@ impl PyBytes {
...
     }
 
+    #[pymethod(name = "__bytes__")]
+    pub fn bytes(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyObjectRef {
+        if zelf.is(&vm.ctx.types.bytes_type) {
+            zelf.into_object()
+        } else {
+            PyBytes::from(zelf.inner.elements.to_vec()).into_object(vm)
+        }
+    }
```
Comment (DimitrisJim) [inline vm/src/builtins/bytes.rs:138]: Don't think you can avoid the clone here.

```suggestion
    #[pymethod(magic)]
    fn bytes(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyRef<PyBytes> {
        if zelf.is(&vm.ctx.types.bytes_type) {
            zelf
        } else {
            PyBytes::from(zelf.inner.clone()).into_ref(vm)
        }
    }
```

**Review by DimitrisJim — CHANGES_REQUESTED** [review]

File: vm/src/builtins/bytes.rs (hunk 2 above)
Comment (Snowapril) [reply vm/src/builtins/bytes.rs:138]: You are right. It seems extremely better than original. Thanks. I sent v3 code.

**Review by youknowone — APPROVED** [review]

**Pull request merged**